internal/alloydb/instance.go

```diff
@@ -220,6 +214,17 @@ func (i *Instance) result(ctx context.Context) (*refreshOperation, error) {
 	return res, nil
 }
 
+// refreshDuration returns the duration to wait before starting the next
+// refresh. Certificates that expire in an hour or less are refreshed in 55
+// minutes. Otherwise, the duration is half of the time until the expiration.
+func refreshDuration(now, certExpiry time.Time) time.Duration {
+	d := certExpiry.Sub(now)
+	if d <= time.Hour {
+		return 55 * time.Minute
+	}
+	return time.Until(now.Add(time.Duration(d / 2)))
```

Comment:
I don't think this is quite what we want to do. If something odd happens and a certificate with a duration of less than 1 hour incurs, this will still wait 55 minutes to refresh that cert.

What we really want the logic to be is in the certExpiry is <= 1hr (although maybe ~65 minutes gives us a better margin of error) then we use a window of 5 minutes. 

How about something like this:
```go
// default window is half of the certs expected duration
w := time.Until(certExpiry) / 2
if w < time.Hour {
    w = 5 * time.Minute
}
return w
```

Reply:
I've extended the edge case checking.